xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift

```diff
@@ -108,33 +114,139 @@ struct UserProfileView: View {
                 )
             }
         })
-        .sheet(
-            isPresented: Binding(
-                get: { state.isMetaSheetPresented },
-                send: send,
-                tag: UserProfileDetailAction.presentMetaSheet
-            )
-        ) {
-            UserProfileDetailMetaSheet(
-                state: state.metaSheet,
-                profile: state,
-                isFollowingUser: state.isFollowingUser,
-                send: Address.forward(
+        .metaSheet(state: state, send: send)
+        .follow(state: state, send: send)
+        .unfollow(state: state, send: send)
+    }
+}
+
+private extension View {
+    func unfollow(state: UserProfileDetailModel, send: @escaping (UserProfileDetailAction) -> Void) -> some View {
+      self.modifier(UnfollowModifier(state: state, send: send))
+    }
+    
+    func follow(state: UserProfileDetailModel, send: @escaping (UserProfileDetailAction) -> Void) -> some View {
+      self.modifier(FollowModifier(state: state, send: send))
+    }
+    
+    func metaSheet(state: UserProfileDetailModel, send: @escaping (UserProfileDetailAction) -> Void) -> some View {
+      self.modifier(MetaSheetModifier(state: state, send: send))
+    }
+}
```

Comment:
I had too many `.sheet`, `.alert` and `.confirmationDialog`s chained on this view for the typechecker, so I broke it into subcomponents via `ViewModifier`s.